src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -3509,6 +3509,57 @@ void StubGenerator::generate_libm_stubs() {
   }
 }
 
+/**
+*  Arguments:
+*
+*  Input:
+*    c_rarg0   - float16  jshort
+*
+*  Output:
+*       xmm0   - float
+*/
+address StubGenerator::generate_float16ToFloat() {
+  StubCodeMark mark(this, "StubRoutines", "float16ToFloat");
+
+  address start = __ pc();
+
+  BLOCK_COMMENT("Entry:");
+  __ enter(); // required for proper stackwalking of RuntimeStub frame
+
+  // Load value into xmm0 and convert
+  __ flt16_to_flt(xmm0, c_rarg0);
+
+  __ leave(); // required for proper stackwalking of RuntimeStub frame
+  __ ret(0);
+
```

Comment:
Do we really need to set up a stack frame for these two?  This should be a leaf, and we have other leaf stubs that don't set up a frame.

Reply:
I think you are right. These stubs are not called from compiled code, only from C++ (C2) code during compilation.
Let me test it.

Reply:
Testing passed with `enter()` and `leave()` removed